components/datadog/apps/fakeintake/fargateFakeintakeService.go

```diff
@@ -0,0 +1,201 @@
+				Protocol:   pulumi.StringPtr("HTTP"),
+				TargetType: pulumi.StringPtr("ip"),
+				VpcId:      pulumi.StringPtr(e.DefaultVPCID()),
+			},
+			Listener: &lb.ListenerArgs{
+				Port:     pulumi.IntPtr(port),
+				Protocol: pulumi.StringPtr("HTTP"),
+			},
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		instance.Host = alb.LoadBalancer.DnsName()
+		balancerArray = classicECS.ServiceLoadBalancerArray{
+			&classicECS.ServiceLoadBalancerArgs{
+				ContainerName:  pulumi.String(containerName),
+				ContainerPort:  pulumi.Int(port),
+				TargetGroupArn: alb.DefaultTargetGroup.Arn(),
+			},
+		}
+	} else {
+		instance.Host, err = FargateServiceFakeintakeWithoutLoadBalancer(e)
+		if err != nil {
+			return nil, err
+		}
+		balancerArray = classicECS.ServiceLoadBalancerArray{}
+	}
+
+	if _, err := ecs.NewFargateService(e.Ctx, namer.ResourceName("srv"), &ecs.FargateServiceArgs{
+		Cluster:              pulumi.StringPtr(e.ECSFargateFakeintakeClusterArn()),
+		Name:                 e.CommonNamer.DisplayName(255, pulumi.String("fakeintake")),
+		DesiredCount:         pulumi.IntPtr(1),
+		EnableExecuteCommand: pulumi.BoolPtr(true),
+		NetworkConfiguration: &classicECS.ServiceNetworkConfigurationArgs{
+			AssignPublicIp: pulumi.BoolPtr(false),
+			SecurityGroups: pulumi.ToStringArray(e.DefaultSecurityGroups()),
+			Subnets:        e.RandomSubnets(),
+		},
+		LoadBalancers: balancerArray,
+		TaskDefinitionArgs: &ecs.FargateServiceTaskDefinitionArgs{
+			Containers: map[string]ecs.TaskDefinitionContainerDefinitionArgs{
+				containerName: {
+					Name:        pulumi.String(containerName),
+					Image:       pulumi.String("public.ecr.aws/datadog/fakeintake:latest"),
+					Essential:   pulumi.BoolPtr(true),
+					MountPoints: ecs.TaskDefinitionMountPointArray{},
+					Environment: ecs.TaskDefinitionKeyValuePairArray{},
+					PortMappings: ecs.TaskDefinitionPortMappingArray{
+						ecs.TaskDefinitionPortMappingArgs{
+							ContainerPort: pulumi.Int(port),
+							HostPort:      pulumi.Int(port),
+							Protocol:      pulumi.StringPtr("tcp"),
+						},
+					},
+					VolumesFrom: ecs.TaskDefinitionVolumeFromArray{},
+				},
+			},
+			Cpu:    pulumi.StringPtr("256"),
+			Memory: pulumi.StringPtr("512"),
+			ExecutionRole: &awsx.DefaultRoleWithPolicyArgs{
+				RoleArn: pulumi.StringPtr(e.ECSTaskExecutionRole()),
+			},
+			TaskRole: &awsx.DefaultRoleWithPolicyArgs{
+				RoleArn: pulumi.StringPtr(e.ECSTaskRole()),
+			},
+			Family: e.CommonNamer.DisplayName(255, pulumi.String("fakeintake-ecs")),
+		},
+		ContinueBeforeSteadyState: pulumi.BoolPtr(true),
+	}, opts...); err != nil {
+		return nil, err
+	}
+	if e.DefaultFargateLoadBalancer() {
+		if err := e.Ctx.RegisterResourceOutputs(instance, pulumi.Map{
+			"host": alb.LoadBalancer.DnsName(),
+		}); err != nil {
+			return nil, err
+		}
+	}
+
+	return instance, nil
+}
+
+func FargateLinuxTaskDefinition(e aws.Environment, name string) (*ecs.FargateTaskDefinition, error) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Looks like this could be private too